test.py

```diff
@@ -195,24 +195,25 @@ def assertPublishedMetric(self, mock, key, value, expected_value=1):
 
         self.assertEqual(actual_value, expected_value, message)
 
-        actual_value = calls[0][0][0].value
-        expected_value = value
-        precision = 0
-
-        if isinstance(value, tuple):
-            expected_value, precision = expected_value
-
-        message = '%s: actual %r, expected %r' % (key,
-                                                  actual_value,
-                                                  expected_value)
-
-        if precision is not None:
-            self.assertAlmostEqual(float(actual_value),
-                                   float(expected_value),
-                                   places=precision,
-                                   msg=message)
-        else:
-            self.assertEqual(actual_value, expected_value, message)
+        if expected_value:
```

Comment:
So this looks like a great catch, but I think the unpublished API is still a little bit busted in that it expects a dictionary. We should just make it take a list of keys. Nothing that necessarily needs to be done in this PR, as it would likely involve changing a bunch of test cases.